src/vt/termination/termination.h

```diff
@@ -102,13 +114,27 @@ struct TerminationDetector :
   using SuccessorBagType   = EpochDependency::SuccessorBagType;
   using EpochGraph         = termination::graph::EpochGraph;
   using EpochGraphMsg      = termination::graph::EpochGraphMsg<EpochGraph>;
+  using EpochStackType     = EpochStack;
 
   /**
    * \internal \brief Construct a termination detector
    */
   TerminationDetector();
 
-  virtual ~TerminationDetector() {}
+  virtual ~TerminationDetector() {
+    //Pop all extraneous epochs off the stack greater than 1
+      auto stack_size = epoch_stack_.size();
+      while (stack_size > 1) {
+        stack_size = (epoch_stack_.pop(), epoch_stack_.size());
+       }
```

Comment:
The way this is written seems overly clever. The variable `stack_size` isn't used outside the loop control, so the whole thing could be written

```suggestion
      // Pop all extraneous epochs off the stack greater than 1
      while (epoch_stack_.size() > 1) {
        epoch_stack_.pop();
      }
```

Also, the indentation was messed up.

That said, this is not a big concern.

Reply:
You can blame the "cleverness" on me (it was copied from AM).

Reply:
done